File: app/src/main/java/lv/rigadevday/android/ui/venues/VenuesRootFragment.java
package lv.rigadevday.android.ui.venues;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;

import butterknife.Bind;
import lv.rigadevday.android.R;
import lv.rigadevday.android.ui.base.BaseFragment;
import lv.rigadevday.android.ui.base.ViewPagerAdapter;

/**
 */
public class VenuesRootFragment extends BaseFragment {

    @Bind(R.id.venues_tabs)
    protected TabLayout mTabs;
    @Bind(R.id.venues_pager)
    protected ViewPager mPager;

    @Override
    protected int contentViewId() {
        return R.layout.fragment_venues_root;
    }

    @Override
    protected void init(Bundle savedInstanceState) {
        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());

        String conference = getString(R.string.venue_conference);
        String workshops = getString(R.string.venue_workshops);
        String afterparty = getString(R.string.venue_afterparty);
        String hotel = getString(R.string.venue_hotel);

        adapter.addFragment(VenueFragment.newInstance(conference), conference);
        adapter.addFragment(VenueFragment.newInstance(workshops), workshops);
        adapter.addFragment(VenueFragment.newInstance(afterparty), afterparty);
        adapter.addFragment(VenueFragment.newInstance(hotel), hotel);

        mPager.setAdapter(adapter);
        mTabs.setupWithViewPager(mPager);
    }

}

File: app/src/main/kotlin/lv/rigadevday/android/ui/venues/VenuesRootFragment.kt
package lv.rigadevday.android.ui.venues

import android.view.View
import kotlinx.android.synthetic.main.fragment_venues_root.view.*
import lv.rigadevday.android.R
import lv.rigadevday.android.ui.base.BaseFragment
import lv.rigadevday.android.ui.base.ViewPagerAdapter
import lv.rigadevday.android.utils.BaseApp
import lv.rigadevday.android.utils.showMessage

class VenuesRootFragment : BaseFragment() {

    override val layoutId = R.layout.fragment_venues_root

    private var adapter: ViewPagerAdapter? = null

    override fun inject() {
        BaseApp.graph.inject(this)
    }

    override fun viewReady(view: View) {
        setupActionBar(R.string.venues_title)

        adapter = ViewPagerAdapter(childFragmentManager)

        var index = 0

        dataFetchSubscription = repo.venues().subscribe(
                {
                    adapter?.addFragment(VenueFragment.newInstance(index), it.name)
                    index++
                },
                { view.showMessage(R.string.error_message) },
                {
                    view.venues_pager.offscreenPageLimit = 2
                    view.venues_pager.adapter = adapter
                    view.venues_tabs.setupWithViewPager(view.venues_pager)
                }
        )
    }
}